### ISSUE TITLE
KeyError: 'a0d7b954_ftp' when trying to run a update check.

### ISSUE THREAD

**Issue by andyblac** [home-assistant/supervisor#4528]
### Describe the issue you are experiencing

Supervisor logs shows `KeyError: 'a0d7b954_ftp'` when trying to run a update check.

### What type of installation are you running?

Home Assistant OS

### Which operating system are you running on?

Home Assistant Operating System

### Steps to reproduce the issue

1. run an update check
2. go to supervisor logs

### Anything in the Supervisor logs that might be useful for us?

```txt
23-09-05 14:05:36 INFO (MainThread) [supervisor.updater] Fetching update data from https://version.home-assistant.io/stable.json
23-09-05 14:05:39 ERROR (MainThread) [aiohttp.server] Error handling request
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/aiohttp/web_protocol.py", line 433, in _handle_request
    resp = await request_handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/aiohttp/web_app.py", line 504, in _handle
    resp = await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/aiohttp/web_middlewares.py", line 117, in impl
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/supervisor/supervisor/api/middleware/security.py", line 185, in block_bad_requests
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/supervisor/supervisor/api/middleware/security.py", line 201, in system_validation
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/supervisor/supervisor/api/middleware/security.py", line 268, in token_validation
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/supervisor/supervisor/api/middleware/security.py", line 280, in core_proxy
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/supervisor/supervisor/api/utils.py", line 62, in wrap_api
    answer = await method(api, *args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/supervisor/supervisor/api/store.py", line 178, in store_info
    ATTR_ADDONS: [
                 ^
  File "/usr/src/supervisor/supervisor/api/store.py", line 179, in <listcomp>
    self._generate_addon_information(self.sys_addons.store[addon])
  File "/usr/src/supervisor/supervisor/api/store.py", line 114, in _generate_addon_information
    ATTR_ADVANCED: addon.advanced,
                   ^^^^^^^^^^^^^^
  File "/usr/src/supervisor/supervisor/addons/model.py", line 227, in advanced
    return self.data[ATTR_ADVANCED]
           ^^^^^^^^^
  File "/usr/src/supervisor/supervisor/store/addon.py", line 19, in data
    return self.sys_store.data.addons[self.slug]
           ~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^
KeyError: 'a0d7b954_ftp'
23-09-05 14:05:42 INFO (MainThread) [supervisor.store] Loading add-ons from store: 75 all - 0 new - 0 remove
23-09-05 14:05:42 INFO (MainThread) [supervisor.store] Loading add-ons from store: 75 all - 0 new - 0 remove
```
```


### System Health information

## System Information

version | core-2023.8.4
-- | --
installation_type | Home Assistant OS
dev | false
hassio | true
docker | true
user | root
virtualenv | false
python_version | 3.11.4
os_name | Linux
os_version | 6.1.21-v8
arch | aarch64
timezone | Europe/London
config_dir | /config

<details><summary>Home Assistant Community Store</summary>

GitHub API | ok
-- | --
GitHub Content | ok
GitHub Web | ok
GitHub API Calls Remaining | 4982
Installed Version | 1.32.1
Stage | running
Available Repositories | 1281
Downloaded Repositories | 33

</details>

<details><summary>Home Assistant Cloud</summary>

logged_in | false
-- | --
can_reach_cert_server | ok
can_reach_cloud_auth | ok
can_reach_cloud | ok

</details>

<details><summary>Home Assistant Supervisor</summary>

host_os | Home Assistant OS 10.5
-- | --
update_channel | stable
supervisor_version | supervisor-2023.08.1
agent_version | 1.5.1
docker_version | 23.0.6
disk_total | 916.2 GB
disk_used | 8.9 GB
healthy | true
supported | true
board | yellow
supervisor_api | ok
version_api | ok
installed_addons | Mosquitto broker (6.3.0), Zigbee2MQTT (1.33.0-1), InfluxDB (4.7.0), Grafana (9.0.3), Studio Code Server (5.10.1), Advanced SSH & Web Terminal (15.0.7), MariaDB (2.6.1), otmonitor (dev)

</details>

<details><summary>Dashboards</summary>

dashboards | 7
-- | --
resources | 26
views | 24
mode | storage

</details>

<details><summary>Recorder</summary>

oldest_recorder_run | 24 August 2023 at 20:09
-- | --
current_recorder_run | 3 September 2023 at 11:47
estimated_db_size | 330.77 MiB
database_engine | mysql
database_version | 10.6.12

</details>

### Supervisor diagnostics

[config_entry-hassio-090f90913227b9da6a1e08a48ee2c240.json.txt](https://github.com/home-assistant/supervisor/files/12523976/config_entry-hassio-090f90913227b9da6a1e08a48ee2c240.json.txt)


### Additional information

_No response_

**Comment by agners** [comment]
This is most likely addressed by #4519. The fix will be released soon.

**Comment by andyblac** [comment]
great thanks.
